google/auth/_helpers.py

```diff
@@ -271,3 +273,33 @@ def is_python_3():
         bool: True if the Python interpreter is Python 3 and False otherwise.
     """
     return sys.version_info > (3, 0)
+
+
+def hash_sensitive_info(data: dict) -> dict:
+    """
+    Hashes sensitive information within a dictionary.
+
+    Args:
+        data: The dictionary containing data to be processed.
+
+    Returns:
+        A new dictionary with sensitive values replaced by their SHA256 hashes.
+    """
+    hashed_data = {}
+    for key, value in data.items():
+        if key in _SENSITIVE_FIELDS:
+            hashed_data[key] = _hash_value(value, key)
+        else:
+            hashed_data[key] = value
+    return hashed_data
+
+
+def _hash_value(value, field_name: str) -> str:
+    """Hashes a value and returns a formatted hash string."""
+    if value is None:
+        return None
+    encoded_value = str(value).encode("utf-8")
+    hash_object = hashlib.sha256()
```

Comment:
```suggestion
    hash_object = hashlib.sha512()
```

Assuming that this will only be called when debugging, is there a strong reason to use `sha256()` instead of `sha512()` which is more secure?